frontend/src/component/events/EventTimeline/EventTimelineHeader/EventTimelineHeader.tsx

```diff
@@ -0,0 +1,167 @@
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { useEnvironments } from 'hooks/api/getters/useEnvironments/useEnvironments';
+import type { IEnvironment } from 'interfaces/environments';
+import { useEffect, useMemo } from 'react';
+
+const StyledCol = styled('div')(({ theme }) => ({
+    display: 'flex',
+    alignItems: 'center',
+    gap: theme.spacing(1),
+}));
+
+const StyledFilter = styled(TextField)(({ theme }) => ({
+    color: theme.palette.text.secondary,
+    '& > div': {
+        background: 'transparent',
+        '& > .MuiSelect-select': {
+            padding: theme.spacing(0.5, 4, 0.5, 1),
+            background: 'transparent',
+        },
+        '& > fieldset': { borderColor: 'transparent' },
+    },
+}));
+
+export type TimeSpanOption = {
+    key: string;
+    label: string;
+    value: Duration;
+    markers: string[];
+};
+
+export const timeSpanOptions: TimeSpanOption[] = [
+    {
+        key: '30m',
+        label: 'last 30 min',
+        value: { minutes: 30 },
+        markers: ['30 min ago'],
+    },
+    {
+        key: '1h',
+        label: 'last hour',
+        value: { hours: 1 },
+        markers: ['1 hour ago', '30 min ago'],
+    },
+    {
+        key: '3h',
+        label: 'last 3 hours',
+        value: { hours: 3 },
+        markers: ['3 hours ago', '2 hours ago', '1 hour ago'],
+    },
+    {
+        key: '12h',
+        label: 'last 12 hours',
+        value: { hours: 12 },
+        markers: ['12 hours ago', '9 hours ago', '6 hours ago', '3 hours ago'],
+    },
+    {
+        key: '24h',
+        label: 'last 24 hours',
+        value: { hours: 24 },
+        markers: [
+            '24 hours ago',
+            '18 hours ago',
+            '12 hours ago',
+            '6 hours ago',
+        ],
+    },
+    {
+        key: '48h',
+        label: 'last 48 hours',
+        value: { hours: 48 },
+        markers: [
+            '48 hours ago',
+            '36 hours ago',
+            '24 hours ago',
+            '12 hours ago',
+        ],
+    },
+];
+
+interface IEventTimelineHeaderProps {
+    totalEvents: number;
+    timeSpan: TimeSpanOption;
+    setTimeSpan: (timeSpan: TimeSpanOption) => void;
+    environment: IEnvironment | undefined;
+    setEnvironment: (environment: IEnvironment) => void;
+}
+
+export const EventTimelineHeader = ({
+    totalEvents,
+    timeSpan,
+    setTimeSpan,
+    environment,
+    setEnvironment,
+}: IEventTimelineHeaderProps) => {
+    const { environments } = useEnvironments();
+
+    const activeEnvironments = useMemo(
+        () => environments.filter(({ enabled }) => enabled),
+        [environments],
+    );
+
+    useEffect(() => {
+        if (activeEnvironments.length > 0) {
+            const defaultEnvironment =
+                activeEnvironments.find(({ type }) => type === 'production') ||
+                activeEnvironments[0];
+            setEnvironment(defaultEnvironment);
```

Comment:
Here is the logic for the default environment selection, as mentioned in the PR description.